numpy/_core/code_generators/ufunc_docstrings.py

```diff
@@ -4275,6 +4275,127 @@ def add_newdoc(place, name, doc):
     See Also
     --------
     str.isalpha
+
+    """)
+
+add_newdoc('numpy._core.umath', 'isdigit',
+    """
+    Returns true for each element if all characters in the string are
+    digits and there is at least one character, false otherwise.
+
+    For 8-bit strings, this method is locale-dependent.
+
+    Parameters
+    ----------
+    x : array_like, with ``bytes_`` or ``unicode_`` dtype
+    $PARAMS
+
+    Returns
+    -------
+    y : ndarray
+        Output array of bools
+        $OUT_SCALAR_1
+
+    See Also
+    --------
+    str.isdigit
+
+    Examples
+    --------
+    >>> a = np.array(['a', 'b', '0'])
+    >>> np.char.isdigit(a)
+    array([False, False,  True])
+    >>> a = np.array([['a', 'b', '0'], ['c', '1', '2']])
+    >>> np.char.isdigit(a)
+    array([[False, False,  True], [False,  True,  True]])
+
+    """)
+
+add_newdoc('numpy._core.umath', 'isspace',
+    """
+    Returns true for each element if there are only whitespace
+    characters in the string and there is at least one character,
+    false otherwise.
+
+    For 8-bit strings, this method is locale-dependent.
```

Comment:
For byte strings I don't think this is useful, it should check only for ASCII spaces (e.g. the C locale). This follows CPython's implementation for `bytes`. If we really care about backward compatibility we can make the `np.char` variant preserve this behavior.

Can you also add the bits defining exactly which unicode characters count as spaces from the CPython `isspace` docs to this (assuming that applies) - I think it's useful to define that precisely.

Reply:
> For byte strings I don't think this is useful, it should check only for ASCII spaces (e.g. the C locale). This follows CPython's implementation for bytes. If we really care about backward compatibility we can make the np.char variant preserve this behavior.

Yup, that makes sense and, if it were up to me, I'd just keep the new behavior but I'm not sure what the policy around backwards-compitibility stuff like this is. If we replace `np.char` with `np.strings` the new ufunc in `np.strings` should definitely have the CPython behavior.

> Can you also add the bits defining exactly which unicode characters count as spaces from the CPython isspace docs to this (assuming that applies) - I think it's useful to define that precisely.

Good idea. I'll do that.

Reply:
```
bytes.isspace()

Return True if all bytes in the sequence are ASCII whitespace and the sequence is not empty, False otherwise. ASCII whitespace characters are those byte values in the sequence b' \t\n\r\x0b\f' (space, tab, newline, carriage return, vertical tab, form feed).
```
Python 2 had some insane definitions, that is where the docs come from.

Note that the current code does **not** match this!  Because the latin1 character `\u00A0` (non breakable space) *is* a space, but is not ascii.  There is some argument to be made that accepting it is OKish (since our byte strings are more string-like), but maybe it's better not (no change in behavior also).

Reply:
> Note that the current code does not match this!

Do you mean the current code in CPython? If so, current CPython main returns False for `\u00A0`, so it does match the documented behavior.

Reply:
I mean your code of course:
```
np.char.isspace(np.array("\u00a0".encode("latin1"), dtype="S"))
```

Reply:
Makes more sense. I'd say let's go ahead and change it to follow the CPython behavior.